service/src/test/java/bio/terra/workspace/service/workspace/flight/CreateGoogleContextFlightTest.java

```diff
@@ -106,29 +110,80 @@ void successCreatesProjectAndContext() throws Exception {
+  void createsProjectAndContext_noBillingAccount_fails() throws Exception {
+    UUID workspaceId = createWorkspace(spendUtils.noBillingAccount());
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    assertTrue(workspaceService.getAuthorizedGcpCloudContext(workspaceId, userRequest).isEmpty());
+
+    // Retry steps once to validate idempotency.
+    Map<String, StepStatus> retrySteps = getStepNameToStepStatusMap();
+    FlightDebugInfo debugInfo = FlightDebugInfo.newBuilder().doStepFailures(retrySteps).build();
+
+    FlightState flightState =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            CreateGcpContextFlight.class,
+            createInputParameters(workspaceId, userRequest),
+            STAIRWAY_FLIGHT_TIMEOUT,
+            debugInfo);
+
+    assertEquals(FlightStatus.ERROR, flightState.getFlightStatus());
+    assertTrue(workspaceService.getAuthorizedGcpCloudContext(workspaceId, userRequest).isEmpty());
+    String projectId =
+        flightState.getResultMap().get().get(WorkspaceFlightMapKeys.GCP_PROJECT_ID, String.class);
+    // The Project should exist, but requested to be deleted.
+    Project project = crl.getCloudResourceManagerCow().projects().get(projectId).execute();
+    assertEquals(projectId, project.getProjectId());
+    assertEquals("DELETE_REQUESTED", project.getState());
+  }
+
+  @Test
+  @DisabledIfEnvironmentVariable(named = "TEST_ENV", matches = BUFFER_SERVICE_DISABLED_ENVS_REG_EX)
+  void createsProjectAndContext_emptySpendProfile_fails() throws Exception {
+    UUID workspaceId = createWorkspace(null);
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    assertTrue(workspaceService.getAuthorizedGcpCloudContext(workspaceId, userRequest).isEmpty());
+
+    // Retry steps once to validate idempotency.
+    Map<String, StepStatus> retrySteps = getStepNameToStepStatusMap();
+    FlightDebugInfo debugInfo = FlightDebugInfo.newBuilder().doStepFailures(retrySteps).build();
+
+    FlightState flightState =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            CreateGcpContextFlight.class,
+            createInputParameters(workspaceId, userRequest),
+            STAIRWAY_FLIGHT_TIMEOUT,
+            debugInfo);
+
+    assertEquals(FlightStatus.ERROR, flightState.getFlightStatus());
+    assertTrue(workspaceService.getAuthorizedGcpCloudContext(workspaceId, userRequest).isEmpty());
+    String projectId =
+        flightState.getResultMap().get().get(WorkspaceFlightMapKeys.GCP_PROJECT_ID, String.class);
+    // The Project should exist, but requested to be deleted.
+    Project project = crl.getCloudResourceManagerCow().projects().get(projectId).execute();
+    assertEquals(projectId, project.getProjectId());
+    assertEquals("DELETE_REQUESTED", project.getState());
+  }
+
```

Comment:
You removed a test from `WorkspaceServiceTest` which validated that an unauthorized user could not link a spend profile, can you add a test covering the same case here?

Per our [test user documentation](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/integration/src/main/resources/testusers/README.md) the user `liam.dragonmaw` does not have spend profile access, so you could use that account. You should be able to access those credentials via the `secondaryUser` methods in `UserAccessUtils`

Reply:
thanks for the links!